[Feature/web-company-crud] 업체 생성, 조회 API 추가

ongsttt52 opened the pull request:
### Summary

### Description

<!-- 구체적인 작업 내용, 필요시 이미지 첨부 -->
- 업체 생성 API 추가(createCompany)
- 업체 조회 API 추가(getCompany)
- hibernate가 데이터베이스를 인식할 수 없는 오류가 발생해 web 모듈의 resources에 .env 파일 추가

### Discussion

<!-- 추후 논의할 점 -->
- 
-

### Issue Number or Link

<!-- 예시: closes #issue-number -->

ongsttt52 committed:
feat: 업체 생성 API 추가

-[] Company Entity, DTO 추가
-[] createCompany() 추가

ongsttt52 committed:
feat: 업체 조회 API 추가

-[] getCompany() 추가
-[] createCompany(), getCompany() 가 공통 응답 객체를 반환하도록 수정

kimchijinju commented on charlee-web/src/main/java/com/efa73/charleeweb/company/interfaces/controller/CompanyController.java:
```diff
@@ -0,0 +1,43 @@
+        Company company = companyService.createCompany(CompanyRequest.createEntity(companyRequest));
+        CompanyResponse response = CompanyResponse.createResponse(company);
+
+        return ResponseEntity.created(URI.create("/company/" + company.getId())).body(Api.of(response));
```
/api/company가 되어야 할 것 같습니다.

kimchijinju replied:
체이닝이 두 번 이상 지속된다면
ResponseEntity.created(URI.create("/company/" + company.getId()))
&nbsp; .body(Api.of(response));

위처럼 개행으로 한번 끊어가면 어떨까요?

kimchijinju commented on charlee-web/src/main/java/com/efa73/charleeweb/company/domain/entity/Company.java:
```diff
@@ -0,0 +1,35 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private String name;
```
Column() 어노테이션을 entity 모든 변수에 붙일지, 말지 합의가 필요합니다.

kimchijinju replied:
필요한 부분에만 붙이는거로 합시다.

kimchijinju commented on charlee-web/src/main/java/com/efa73/charleeweb/company/domain/entity/Company.java:
```diff
@@ -0,0 +1,35 @@
+
+    private String name;
+
+    private String siteLink;
```
이렇게 되면 DB에 카멜 케이스로 들어가게 됩니다. 이부분도 합의가 필요합니다

ongsttt52 committed:
feat: 업체 수정 API 추가

-[] updateCompany 추가
-[] 조회 실패했을 경우 예외처리 필요

ongsttt52 committed:
refactor: @Comlumn 어노테이션 추가

-[] Company Entity의 name 컬럼에 @Column(nullable=false) 추가

ongsttt52 committed:
test: DB 테스트 - 시동 ON 정보 저장 테스트 추가

-[] CollectRepositoryTest - saveEventInfo() 추가
-[] 테스트 목적 - 시동 ON 이벤트가 DB에 잘 저장되는지 확인
-[] 현재는 시동 OFF 이벤트와 구분이 필요하지 않을 것으로 생각해 따로 작성하지 않음

yyysolhhh commented on charlee-web/src/main/java/com/efa73/charleeweb/company/domain/service/CompanyService.java:
```diff
@@ -0,0 +1,47 @@
+    public Company updateCompany(CompanyRequest companyRequest, Long companyId) {
+
+        if (companyRepository.existsById(companyId)) {
+            var company = Company.updateEntity(companyId, companyRequest.name(), companyRequest.siteLink());
```
var 대신 타입을 명시적으로 쓰는게 어떨까요?

yyysolhhh reviewed (commented):
수고하셨습니다.

cyunlee commented on charlee-collector/src/test/java/com/efa73/charleecollector/domain/repository/CollectRepositoryTest.java:
```diff
@@ -52,6 +52,17 @@ void saveCycleInfo() {
+    @DisplayName("시동 ON EventInfo가 DB에 잘 저장되는지 확인")
+    void saveEventInfo() {
+
+        var saved = cycleInfoRepository.save(eventInfo);
```
var를 쓴 특별한 이유가 있나요?

cyunlee commented on charlee-web/src/main/java/com/efa73/charleeweb/company/domain/service/CompanyService.java:
```diff
@@ -0,0 +1,47 @@
+        }
+    }
+
+    public void deleteCompany(Long companyId) {
```
if-else 구문에서 else를 사용하는 것보다 조건 불만족시 바로 예외를 던지는 패턴을 사용하면 어떨까요

```java
public void deleteCompany(Long companyId) {
    if (!companyRepository.existsById(companyId) {
        throw new EntityNotFoundException(CustomErrorCode.ENTITY_NOT_FOUND.getMessage());    
   }
   companyRepository.deleteId(companyId);
}
```

cyunlee commented on charlee-web/src/main/java/com/efa73/charleeweb/company/interfaces/dto/CompanyRequest.java:
```diff
@@ -0,0 +1,10 @@
+package com.efa73.charleeweb.company.interfaces.dto;
```
dto 패키지에 request, response 패키지 추가해주세요

cyunlee reviewed (changes requested):
pr 리뷰 상태 수정

ongsttt52 committed:
refactor: Company Update 팩토리 메서드 수정 및 서비스 로직 변경

-[] Company Entity의 팩터리 메서드 updateCompany() 를 static에서 인스턴스 메서드로 변경
-[] 그에 따른 CompanyService 의 updateCompany() 로직 수정
-[] 단위테스트가 필요할 정도의 로직이 없다고 판단, CompanyServiceTest 삭제
-[] 대신 도메인 테스트 CompanyTest 추가
